doc/manual/source/physics/active_particles.rst

```diff
@@ -0,0 +1,146 @@
+.. _active_particles:
+
+
+Active Particles
+============================================
+
+Active Particles (APs) were introduced into Enzo in version 2.6. They were originally part
+of the Enzo-3.0 effort and were developed by a number of authors including Nathan Goldbaum,
+Matt Turk, John Wise, John Regan, Oliver Hahn, Greg Meece, Brian Crosby and others. Their design philosophy was that the
+APs would be highly extensible, have robust feedback capabilities and be very much object orientated.
+They were designed to replace the star object class.
+
+Compared to the star objects the APs can more easily connect with the radiative trasfer solver this making the
+framework more extensible in that context. Other feedback mechanisms (e.g. mechanical and thermal) are
+also easily implemented. 
+
+Using Active Particles
+______________________
+
+To implement the active particles framework in any Enzo run is very simple. Add the following line to the
+parameter file that you are using to (re)start Enzo.
+``AppendActiveParticleType = <YourActiveParticleType>``
+where ``<YourActiveParticleType>`` is one of the currently available AP types. 
+
+Types of Active Particles
+_________________________
+
+At the moment there are a number of AP type implementations. However, with the exception of the "SmartStar"
+AP type none of the implementations have been (robustly) tested.
+
+* SmartStar
+* Accreting Particle
+* Cen-Ostriker
+* GalaxyParticle
+* Kravtsov
+* PopIII
+* RadiationParticle
+* SpringelHernquist
+* Skeleton
+
+The ``Skeleton`` particle is simply an example particle heavily commented so allow a user/developer to develop their
+own AP. All the other particles with the exception of the ``AccretingParticle`` and the ``SmartStar`` particle were ported from the
+Star Object particle implementation described in :ref:`star_particles`.
+
+
+Current Limitations of Active Particles
+_______________________________________
+
+Most of the AP types have not been tested - though in principle do work correctly. Caution and some testing of the particle
+types is advisable for now.
+
+**Multiple AP types can not currently be run together.** This isn't a fundamental limitation. In principle multiple APs can work
+together without difficulty. Some communication work needs to be undertaken to make this work. 
```

Comment:
I see the issue primarily related to creation of the buffer list of particles with the Particle Type ID.

Reply:
Probably.